### DISCUSSION TITLE
European Goldfinch and "Chardonneret élégant" are the same I think

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#810] [Q&A]
Hi,

The European Goldfinch and the " Chardonneret élégant" both are "Carduelis carduelis". They both appeared in my "Recordings" today. In settings I selected the French localisation, 6k model.

![bziWRrO](https://user-images.githubusercontent.com/83965802/228675736-32054629-49cc-4105-bb2e-23df9e4afbc9.png)

**Comment by ehpersonal38** [comment]
Did you temporarily switch your database language to English?

**Reply by blogdemoi** [reply]
I don't think so. I reinstalled everything this morning from scratch, so at first the database was in English then I switched to French a few minutes later.

**Reply by blogdemoi** [reply]
Maybe it is better to wait for tomorrow, to see what happens, as I have a lot of "Chardonneret élégant". :-)

**Comment by blogdemoi** [comment]
I didn't have any "European Goldfinch" detection today. So I suppose you were right @ehpersonal38,  maybe I temporarily switched to English.

However I have a lot of " Chardonneret élégant" / "Carduelis carduelis" (the same specie as "European Goldfinch"). When I say a lot you can see by yourself:

![higG5oF](https://user-images.githubusercontent.com/83965802/228960335-537d734d-8e45-4049-84ac-fc74b43531c7.png)

I suspect that a colony may live just outside my window :-) and that is the reason why my system is always a few hours behind. What would be the best solution? Put this specie in the excluded species list? But in that case they won't appear in the detections...?

Edit: sadly an Eurasian collared dove was killed today at my home by my neighbour's cat :-(

**Reply by ehpersonal38** [reply]
That's interesting that your analysis system can't keep up with the demand... didn't you have this problem a month or so ago too? What exactly is happening? On my Pi 4, analysis takes about 1 second, and with recordings being 15 seconds this leaves plenty of time so that a backlog never occurs. On your system your analysis time must be > than your recording length, is that right?

I'm sorry to hear about that dove though. 🙁

**Reply by blogdemoi** [reply]
Yes, I have always had this problem, I don't know why. I suspected the Pi 3B not being powerful enough and bought a LePotato. It solved this problem for almost 2 weeks but now it is back. Could it be related to the high number of "European Goldfinch" detections. They do sing all day long, since a few weeks. 

> analysis takes about 1 second, and with recordings being 15 seconds this leaves plenty of time so that a backlog never occurs. On your system your analysis time must be > than your recording length, is that right?

I will check this tomorrow during the day.

**Reply by blogdemoi** [reply]
>On your system your analysis time must be > than your recording length, is that right?

When I (just) look at my  "View log" I have this:

> 23:21:29---9.0;12.0-('Erithacus rubecula_Rougegorge familier', 0.21187726)
> 23:21:29---12.0;15.0-('Turdus viscivorus_Grive draine', 0.50283855)
> 23:21:29---15.0;18.0-('Turdus merula_Merle noir', 0.40342802)
> 23:21:29---18.0;21.0-('Ficedula hypoleuca_Gobemouche noir', 0.11939934)
> 23:21:29---21.0;24.0-('Turdus merula_Merle noir', 0.12543051)
> 23:21:29---24.0;27.0-('Turdus merula_Merle noir', 0.16160779)
> 23:21:29---27.0;30.0-('Turdus merula_Merle noir', 0.24549358)
> 23:21:29---RECORDING_LENGTH set to 30
> 23:21:30---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts/analyze.py --i BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:10:43.wav --o BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:10:43.wav.csv --lat 49.XX --lon 2.XX --week 12 --overlap 0.0 --sensitivity 1.25 --min_conf 0.75 --include_list BirdNET-Pi/include_species_list.txt --exclude_list BirdNET-Pi/exclude_species_list.txt --birdweather_id "IN_USE"
> 23:21:30---ANALYZING AUDIO... DONE! Time 5.1 SECONDS
> 23:21:30---WRITING RESULTS TO BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:10:13.wav.csv ... DONE! WROTE 0 RESULTS.
> 23:21:30---READING AUDIO DATA... DONE! READ 10 CHUNKS.
> 23:21:30---Date-time: 2023-03-30 16:10:43
> 23:21:35---0.0;3.0-('Serinus serinus_Serin cini', 0.061669007)
> 23:21:35---3.0;6.0-('Turdus merula_Merle noir', 0.32614532)
> 23:21:35---6.0;9.0-('Turdus merula_Merle noir', 0.26374075)
> 23:21:35---9.0;12.0-('Ficedula hypoleuca_Gobemouche noir', 0.09956491)
> 23:21:35---12.0;15.0-('Turdus merula_Merle noir', 0.36165792)
> 23:21:35---15.0;18.0-('Ficedula hypoleuca_Gobemouche noir', 0.19359885)
> 23:21:35---18.0;21.0-('Turdus merula_Merle noir', 0.2797583)
> 23:21:35---21.0;24.0-('Turdus merula_Merle noir', 0.3243402)
> 23:21:35---24.0;27.0-('Turdus merula_Merle noir', 0.18265428)
> 23:21:35---27.0;30.0-('Turdus merula_Merle noir', 0.44913417)
> 23:21:35---RECORDING_LENGTH set to 30
> 23:21:35---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts/analyze.py --i BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:11:13.wav --o BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:11:13.wav.csv --lat 49.XX --lon 2.XX --week 12 --overlap 0.0 --sensitivity 1.25 --min_conf 0.75 --include_list BirdNET-Pi/include_species_list.txt --exclude_list BirdNET-Pi/exclude_species_list.txt --birdweather_id "IN_USE"
> 23:21:36---ANALYZING AUDIO... DONE! Time 5.0 SECONDS
> 23:21:36---WRITING RESULTS TO BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:10:43.wav.csv ... DONE! WROTE 0 RESULTS.
> 23:21:36---READING AUDIO DATA... DONE! READ 10 CHUNKS.
> 23:21:36---Date-time: 2023-03-30 16:11:13
> 23:21:41---0.0;3.0-('Ficedula hypoleuca_Gobemouche noir', 0.084815785)
> 23:21:41---3.0;6.0-('Turdus merula_Merle noir', 0.3460208)
> 23:21:41---6.0;9.0-('Turdus merula_Merle noir', 0.09523814)
> 23:21:41---9.0;12.0-('Pica pica_Pie bavarde', 0.109659694)
> 23:21:41---12.0;15.0-('Buteo buteo_Buse variable', 0.1336259)
> 23:21:41---15.0;18.0-('Turdus viscivorus_Grive draine', 0.09129031)
> 23:21:41---18.0;21.0-('Turdus merula_Merle noir', 0.22668992)
> 23:21:41---21.0;24.0-('Turdus merula_Merle noir', 0.22320637)
> 23:21:41---24.0;27.0-('Turdus merula_Merle noir', 0.15958188)
> 23:21:41---27.0;30.0-('Turdus merula_Merle noir', 0.30877343)
> 23:21:41---RECORDING_LENGTH set to 30
> 23:21:42---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts/analyze.py --i BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:11:43.wav --o BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:11:43.wav.csv --lat 49.XX --lon 2.XX --week 12 --overlap 0.0 --sensitivity 1.25 --min_conf 0.75 --include_list BirdNET-Pi/include_species_list.txt --exclude_list BirdNET-Pi/exclude_species_list.txt --birdweather_id "IN_USE"
> 23:21:42---ANALYZING AUDIO... DONE! Time 5.1 SECONDS
> 23:21:42---WRITING RESULTS TO BirdSongs/March-2023/30-Thursday/2023-03-30-birdnet-16:11:13.wav.csv ... DONE! WROTE 0 RESULTS.
> 23:21:42---READING AUDIO DATA... DONE! READ 10 CHUNKS.

**Reply by ehpersonal38** [reply]
So from that log, your recording length is 30 seconds, and it takes on average 5 seconds to process a recording. Meaning you theoretically can process 6 files without starting a backlog, yet your system is struggling to even process 1 file in time.

I'm not sure exactly what this means, maybe at some point your processing time is shooting up from ~5s to ~30s+ randomly. Maybe you can watch to see if that happens, and then run `htop` to see what exactly your system is doing that's causing taking all that processing power.

**Reply by blogdemoi** [reply]
After a few days with the "European Goldfinch"  in the exclusion list I can tell you that my system is back to normal. Too many detections of "European Goldfinch" I presume...
